### PR TITLE
Change verify-production-urls to work when the pipeline has already uploaded some files to S3

### PR DESCRIPTION
**Pull request by Dantemss**
For: https://github.com/openstax/unified/issues/1808
Goes with: https://github.com/openstax/unified-deployment/pull/240
Necessary because some of the files are stored in S3 and not sent back to the main Concourse worker

### CODE REVIEW

File: concourse/verify-production-urls/script.bash [hunk 1]
```diff
@@ -1,23 +1,64 @@
...
+echo "found $new_release_prod_number production urls in new release $REACT_APP_RELEASE_ID"
+
+missing_prod_files=$(diff_file_lists "$current_prod_files" "$new_release_prod_files")
+
+if [ -n "$missing_prod_files" ]; then
+  printf "MISSING PRODUCTION PATHS:\n%s\n" "$missing_prod_files"
+  exit 1
+fi
+
+new_release_sand_files=$(get_bucket_release_files sandbox "$REACT_APP_RELEASE_ID")
```
Comment (TomWoodward) [inline concourse/verify-production-urls/script.bash:52]: lets have this script only validate the production files, i would like to work towards not having them all duplicated in sandbox anyway

**Review by TomWoodward — CHANGES_REQUESTED** [review]

File: concourse/verify-production-urls/script.bash [hunk 2]
```diff
@@ -1,23 +1,64 @@
...
-done
+# Bash array difference (with newline-separated inputs)
+# Adapted from https://stackoverflow.com/a/28161520
+# Returns files in $prod_files but not in $uploaded_files
+# Algorithm:
+# 1. Concatenate all entries, newline-separated
+# 2. Sort entries
+# 3. Remove all lines that appear more than once
+# The array to be removed appears twice in the input so none of its entries are added to the output
+diff_file_lists () { printf "%s\n%s\n%s\n" "$1" "$2" "$2" | sort | uniq -u; }
```
[the remark this answers is not in the document]
Comment (Dantemss) [reply concourse/verify-production-urls/script.bash:32]: It didn't seem that bad, only took about a second on my machine.

[the remark this answers is not in the document]
Comment (Dantemss) [reply concourse/verify-production-urls/script.bash:23]: This diff takes 0.45second on my machine. However, I have found a faster implementation using `comm` instead that takes only 0.15second. I'll update.

File: concourse/verify-production-urls/script.bash (hunk 1 above)
Comment (Dantemss) [reply concourse/verify-production-urls/script.bash]: Done

**Review by TomWoodward — APPROVED** [review]

**Pull request merged**